Fix broken/missing reset after no admin reaction

The-personified-devil opened the pull request:
Fix an issue in the massjoin detection where important reset/cleanup code
wouldn't be run, because `wait_for()` threw an exception since no admin reacted.
To combat this, a try...finally was added.

Fixes #31

Commit:
Fix broken/missing reset after no admin reaction

Fix an issue in the massjoin detection where important reset/cleanup code
wouldn't be run, because `wait_for()` threw an exception since no admin reacted.
To combat this, a try...finally was added.

Signed-off-by: Leonhard Saam <leonhard.saam@yahoo.com>

Yuhanun commented on src/cogs/rules.py:
```diff
@@ -81,70 +81,73 @@ async def on_member_join(self, member):
 
-            self._recent_joins.clear()
-            self.massjoin_active = False
+            finally:
```
Should probably catch it and send the exception if it occurs so we know whether to manually ban or not?

The-personified-devil replied:
This was mostly intended for just making sure the reset code is run.
With the main reason being the actually relatively likely to fail `wait_for()`s and everything else going properly being an assumption.

So there's no logic for ensuring that anything special going wrong being handled as
- when the actual banning fails, the success message doesn't appear
- when the notification message isn't sent, the admins will notice anyway
- when the ban "wizard" message[s] aren't sent, the admins will notice that too and can take action manually

Yuhanun replied:
I'd like visual feedback if the banning fails, the rest is fine

Commit:
Fix issues with the massjoin messages

- Fix the reaction emojis being missing from the config file
- Fix message string not being a format string

Signed-off-by: Leonhard Saam <leonhard.saam@yahoo.com>

Commit:
Make user banning more verbose

- Add failure notice
- Add starting message

Signed-off-by: Leonhard Saam <leonhard.saam@yahoo.com>

Commit:
Move message chunking into helper function

Also:
- Add support for messages without newlines to split on
- Add helper function to send chunks

Signed-off-by: Leonhard Saam <leonhard.saam@yahoo.com>

Commit:
Make failed bans more verbose and ban greedily

- Send the exception
- List users that failed to be banned
- Try to ban as many people as possible by trying the individual ban and
  not the entire loop

Also:
- Remove unneeded import of the io module
- Remove additional new line after `on_member_join(...)`
- Use parentheses in the banning start message

Signed-off-by: Leonhard Saam <leonhard.saam@yahoo.com>
